bugsnag-android-performance/src/test/java/com/bugsnag/android/performance/internal/processing/TracerTest.kt

```diff
@@ -19,11 +21,16 @@ import org.robolectric.RobolectricTestRunner
 class TracerTest {
     private lateinit var tracer: Tracer
     private lateinit var spanFactory: SpanFactory
+    private val spanEndCallback1 = SpanEndCallback { true }
+    private val spanEndCallback2 = SpanEndCallback { false }
+    private val worker = mock<Worker>()
 
     @Before
     fun createTracer() {
-        tracer = Tracer()
+        tracer = Tracer(emptyArray())
+        tracer.worker = worker
         spanFactory = SpanFactory(tracer)
+        tracer.spanEndCallbacks = mutableListOf(spanEndCallback1)
```

Comment:
I think that this can maybe be refactored so that the `spanEndCallbacks` don't need to be `var`, instead constructing the test `Tracer` on a per-test basis.